core/domain/exp_services.py

```diff
@@ -1795,3 +1799,203 @@ def save_multi_exploration_math_rich_text_info_model(
+
+    Raises:
+        Exception: If any of the SVG images provided fail validation.
+    """
+    exploration = exp_fetchers.get_exploration_by_id(exp_id)
+    html_strings_after_conversion = ''
+    change_lists = []
+    # Create a dict with dimensions for each LaTeX string. This dict will be
+    # used for generating a filename for each raw_latex string in the HTML.
+    raw_latex_to_dimensions_dict = {}
+    for raw_latex, raw_latex_dict in image_data.items():
+        raw_latex_to_dimensions_dict[raw_latex] = {}
+        raw_latex_to_dimensions_dict[raw_latex]['dimensions'] = (
+            raw_latex_dict['dimensions'])
+
+    for state_name, state in exploration.states.items():
+        add_svg_filenames_for_latex_strings_in_html_string = (
+            functools.partial(
+                html_validation_service.
+                add_svg_filenames_for_latex_strings_in_html_string,
+                raw_latex_to_dimensions_dict))
+        old_state_dict = copy.deepcopy(state.to_dict())
+        converted_state_dict = (
+            state_domain.State.convert_html_fields_in_state(
+                state.to_dict(),
+                add_svg_filenames_for_latex_strings_in_html_string))
+        converted_state = state_domain.State.from_dict(converted_state_dict)
+        html_strings_after_conversion += (
+            ''.join(converted_state.get_all_html_content_strings()))
+        change_lists.extend(
+            generate_html_change_list_for_state(
+                state_name, converted_state_dict, old_state_dict))
+
+    filenames_mapping = (
+        html_validation_service.
+        extract_svg_filename_latex_mapping_in_math_rte_components(
+            html_strings_after_conversion))
+
+    for filename, raw_latex in filenames_mapping:
+        image_file = image_data[raw_latex]['file']
+        image_validation_error_message_suffix = (
+            'SVG image provided for latex %s failed validation' % raw_latex)
+        if not image_file:
+            logging.error(
+                'Image not provided for filename %s' % (filename))
+            raise Exception(
+                'No image data provided for file with name %s. %s'
+                % (filename, image_validation_error_message_suffix))
+        try:
+            file_format = (
+                image_validation_services.validate_image_and_filename(
+                    image_file, filename))
+        except utils.ValidationError as e:
+            e = '%s %s' % (e, image_validation_error_message_suffix)
+            raise Exception(e)
+        image_is_compressible = (
+            file_format in feconf.COMPRESSIBLE_IMAGE_FORMATS)
+        fs_services.save_original_and_compressed_versions_of_image(
+            filename, feconf.ENTITY_TYPE_EXPLORATION, exp_id, image_file,
+            'image', image_is_compressible)
+
+
+    update_exploration(
+        user_id, exp_id, change_lists, 'added SVG images for math tags.')
```

Comment:
This will cause the user_id to see the exploration on their list of created/edited explorations. Is that actually desirable? Should it be attributed to a bot instead?

Reply:
changed this to migration bot, is that fine?